eureka-client/src/main/java/com/netflix/discovery/converters/wrappers/DecoderWrapper.java

```diff
@@ -0,0 +1,14 @@
+package com.netflix.discovery.converters.wrappers;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * @author David Liu
+ */
+public interface DecoderWrapper extends CodecWrapperBase {
```

Comment:
Either way works. Once we move beyond the transitions phase and deprecate the older codecs, the wrappers can be removed and the actual codec classes be moved to implement the Encoder and Decoder interface.